src/engine/Server.cpp

```diff
@@ -164,31 +164,34 @@ void Server::run(const string& indexBaseName, bool useText, bool usePatterns,
 }
 
 // _____________________________________________________________________________
-ad_utility::UrlParser::ParsedHTTPRequest Server::parseHttpRequest(
-    const ad_utility::httpUtils::HttpRequest auto& request) const {
+ad_utility::UrlParser::ParsedRequest Server::parseHttpRequest(
+    const ad_utility::httpUtils::HttpRequest auto& request) {
   // For an HTTP request, `request.target()` yields the HTTP Request-URI.
   // This is a concatenation of the URL path and the query strings.
-  ad_utility::UrlParser::ParsedHTTPRequest parsedUrl =
-      ad_utility::UrlParser::parseRequestTarget(request.target());
+  auto parsedUrl = ad_utility::UrlParser::parseRequestTarget(request.target());
+  ad_utility::UrlParser::ParsedRequest parsedRequest{
+      std::move(parsedUrl.path_), std::move(parsedUrl.parameters_),
+      std::nullopt};
 
   if (request.method() == http::verb::get) {
-    // There can be valid GET requests without a query. E.g. the command to get
-    // index statistics.
-    if (parsedUrl.parameters_.contains("query")) {
-      parsedUrl.query_ = parsedUrl.parameters_["query"];
+    // Requests for QLever's custom commands (e.g. retrieving index statistics)
+    // don't have a query.
+    if (parsedRequest.parameters_.contains("query")) {
+      parsedRequest.query_ = parsedRequest.parameters_["query"];
```

Comment:
These five lines (including the comment) are duplicated below.
Please extract them to a lambda `extractQueryFromParameters`.
Also (Not important, but cleaner) should this action also remove the `query` from the `parameters` (not super important, but maybe it's simple).

Reply:
Definitely! I must have gotten sidetracked before I got around to removing the `query` from the `parameters`.